workspaces/liferay-partner-workspace/client-extensions/liferay-partner-etc-spring-boot/src/main/java/com/liferay/partner/QueueListener.java

```diff
@@ -95,20 +91,55 @@ public void listener(
 				return;
 			}
 
-			if (_log.isInfoEnabled()) {
-				_log.info(
-					StringBundler.concat(
-						"Updating account ", salesforceAccountKey,
-						" with name ",
-						koroneikiAccountJSONObject.getString("name")));
+			if (routingKey.equals("koroneiki.account.update")) {
+				if (_log.isInfoEnabled()) {
+					_log.info(
+						StringBundler.concat(
+							"Updating account ", salesforceAccountKey,
+							" with name ",
+							koroneikiAccountJSONObject.getString("name")));
+				}
+
+				String countryISOCode = _getCountryISOCode(
+					koroneikiAccountJSONObject);
+
+				_updateAccount(
+					salesforceAccountKey,
+					koroneikiAccountJSONObject.getString("name"),
+					countryISOCode);
 			}
 
-			String countryISOCode = _getCountryISOCode(
-				koroneikiAccountJSONObject);
+			if (routingKey.equals("koroneiki.account.contactrole.assigned")) {
+				JSONObject contactRoleJSONObject = jsonObject.getJSONObject(
+					"contactRole");
+
+				String contactRoleName = contactRoleJSONObject.getString(
+					"name");
 
-			_updateAccount(
-				salesforceAccountKey,
-				koroneikiAccountJSONObject.getString("name"), countryISOCode);
+				JSONObject contactJSONObject = jsonObject.getJSONObject(
+					"contact");
+
+				_assignUser(
+					koroneikiAccountJSONObject.getString("name"),
+					contactJSONObject.optString("emailAddress"),
+					contactRoleName, salesforceAccountKey);
+			}
+
+			if (routingKey.equals("koroneiki.account.contactrole.unassigned")) {
+				JSONObject contactRoleJSONObject = jsonObject.getJSONObject(
+					"contactRole");
+
+				String contactRoleName = contactRoleJSONObject.getString(
```

Comment:
you can put this directly in the method call. No need to create a variable to be used once